File: rgctl-tests/ecommerce-c/src/models/product.c
#include "ecommerce/models/product.h"
#include <string.h>

void product_init(product_t *obj) { if (obj) { memset(obj, 0, sizeof(*obj)); } }

void product_set_price(product_t *p, double price) { if (p) p->price = price; }

int product_is_valid(const product_t *p) { return p && p->price > 0 && p->name[0]; }

File: rgctl-tests/ecommerce-csharp/src/Ecommerce/Models/Product.cs
namespace Ecommerce.Models;

public class Product
{
    public long Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public long CategoryId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Category Category { get; set; } = null!;
    public Inventory? Inventory { get; set; }
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}
